validation/linux_ns_path.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"github.com/mndrix/tap-go"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/validation/util"
+)
+
+func testNamespacePath(t *tap.T, ns string, unshareOpt string) error {
+	// Calling 'unshare' (part of util-linux) is easier than doing it from
+	// Golang: mnt namespaces cannot be unshared from multithreaded
+	// programs.
+	cmd := exec.Command("unshare", unshareOpt, "--fork", "sleep", "10000")
+	err := cmd.Start()
+	if err != nil {
+		return fmt.Errorf("cannot run unshare: %s", err)
+	}
+	defer func() {
+		if cmd.Process != nil {
+			cmd.Process.Kill()
```

Comment:
The sleep processes are surviving the death of the `unshare` parent unfortunately.

I tried `cmd.Process.Signal(syscall.SIGTERM)` but it does not help because:
- unshare does not catch the signal so it dies without forwarding the SIGTERM to its children
- processes in a different pid namespaces would not receive the signal anyway because of signal rules in pid namespaces (see [man 7 pid_namespaces](http://man7.org/linux/man-pages/man7/pid_namespaces.7.html))